kaldb/src/main/java/com/slack/kaldb/logstore/search/LogIndexSearcherImpl.java

```diff
@@ -111,30 +113,39 @@ public SearchResult<LogMessage> search(
       // This is a useful optimization for indexes that are static.
       IndexSearcher searcher = searcherManager.acquire();
       try {
-        TopFieldCollector topFieldCollector = buildTopFieldCollector(howMany);
-        StatsCollector statsCollector =
+        List<LogMessage> results;
+        Histogram histogram = new NoOpHistogramImpl();
+
+        CollectorManager<StatsCollector, Histogram> statsCollector =
             buildStatsCollector(bucketCount, startTimeMsEpoch, endTimeMsEpoch);
-        Collector collectorChain = MultiCollector.wrap(topFieldCollector, statsCollector);
 
-        searcher.search(query, collectorChain);
-        List<LogMessage> results;
         if (howMany > 0) {
-          ScoreDoc[] hits = topFieldCollector.topDocs().scoreDocs;
+          CollectorManager<TopFieldCollector, TopFieldDocs> topFieldCollector =
+              buildTopFieldCollector(howMany, bucketCount > 0);
+          MultiCollectorManager collectorManager = new MultiCollectorManager(topFieldCollector);
```

Comment:
Can we initialize `collectorManager` only once?

Reply:
Done.